test/evm/hardhat/chain-specific-spokepools/Arbitrum_SpokePool.ts

```diff
@@ -138,4 +172,46 @@ describe("Arbitrum Spoke Pool", function () {
     expect(l2GatewayRouter[functionKey]).to.have.been.calledOnce;
     expect(l2GatewayRouter[functionKey]).to.have.been.calledWith(dai.address, hubPool.address, amountToReturn, "0x");
   });
+
+  it("Bridge tokens to hub pool correctly using the OFT messaging for L2 USDT token", async function () {
+    const l2UsdtSendAmount = BigNumber.from("1234567");
+    const { leaves, tree } = await constructSingleRelayerRefundTree(
+      l2UsdtContract.address,
+      await arbitrumSpokePool.callStatic.chainId(),
+      l2UsdtSendAmount
+    );
+    await arbitrumSpokePool.connect(crossDomainAlias).relayRootBundle(tree.getHexRoot(), mockTreeRoot);
+
+    const oftNativeFee = BigNumber.from(1e9 * 200_000); // 1 GWEI gas price * 200,000 gas cost
+
+    // seed `arbitrumSpokePool` with some eth for native fee payment in OFT logic
+    await ethers.provider.send("hardhat_setBalance", [arbitrumSpokePool.address, oftNativeFee.toHexString()]);
+
+    // set up `quoteSend` return val
+    const msgFeeStruct: MessagingFeeStructOutput = [
+      oftNativeFee, // nativeFee
+      BigNumber.from(0), // lzTokenFee
+    ] as MessagingFeeStructOutput;
+    l2OftMessenger.quoteSend.returns(msgFeeStruct);
+
+    // set up `send` return val
+    const msgReceipt: MessagingReceiptStructOutput = [
+      randomBytes32(), // guid
+      BigNumber.from("1"), // nonce
+      msgFeeStruct, // fee
+    ] as MessagingReceiptStructOutput;
+
+    const oftReceipt: OFTReceiptStructOutput = [l2UsdtSendAmount, l2UsdtSendAmount] as OFTReceiptStructOutput;
+
+    l2OftMessenger.send.returns([msgReceipt, oftReceipt]);
+
+    await arbitrumSpokePool.connect(relayer).executeRelayerRefundLeaf(0, leaves[0], tree.getHexProof(leaves[0]));
+    // Adapter should have approved gateway to spend its ERC20.
+    expect(await l2UsdtContract.allowance(arbitrumSpokePool.address, l2OftMessenger.address)).to.equal(
+      l2UsdtSendAmount
+    );
+
+    // We should have called send on the l2OftMessenger
+    expect(l2OftMessenger.send).to.have.been.calledOnce;
```

Comment:
I like using the [calledWith](https://smock.readthedocs.io/en/latest/fakes.html#asserting-call-arguments-or-value) syntax to check the arguments passed to `send` are as expected